aptos-node/src/lib.rs

```diff
@@ -301,14 +328,60 @@ where
     start(config, Some(log_file), false)
 }
 
+/// Merges node_config with the config override file
+fn merge_nodeconfig_override(
+    node_config: NodeConfig,
+    config_override: serde_yaml::Value,
+) -> NodeConfig {
+    serde_merge::tmerge::<NodeConfig, serde_yaml::Value, NodeConfig>(node_config, config_override)
+        .map_err(|e| anyhow::anyhow!("Unable to merge default config with override. Error: {}", e))
+        .unwrap()
+}
+
 /// Creates a single node test config, with a few config tweaks to reduce
 /// the overhead of running the node on a local machine.
 fn create_single_node_test_config(
     config_path: Option<PathBuf>,
+    config_override_path: Option<PathBuf>,
     enable_lazy_mode: bool,
 ) -> NodeConfig {
-    // Build a single validator network with a generated config
-    let mut node_config = NodeConfig::get_default_validator_config();
+    let mut node_config = match config_override_path {
```

Comment:
If you're doing a lot of `if Ok or if Some` then you might want to consider chaining with map instead of deeply nesting, sometimes it can be more readable

Reply:
ah true yea. slightly more readable now with `map_err`